src/components/app-map/index.vue

```diff
@@ -25,13 +30,57 @@ import MapMapbox from './map-mapbox';
 import MapDrawControl from './map-draw-control';
 import MapNavigationControl from './map-navigation-control';
 import MapLayer from './map-layer';
+import MapSearch from './map-search';
+import wms from '../../lib/mapbox/layers/wms';
+import geoserverUrl from '../../lib/geoserver-url';
+
+const tileSize = 256;
+const wmsLayerDefaultConfig = {
+  request: 'GetMap',
+  width: tileSize,
+  height: tileSize,
+  format: 'image/png',
+  srs: 'EPSG:3857',
+  bbox: '{bbox-epsg-3857}',
+  transparent: 'true',
+  encode: false
+};
 
 export default {
   components: {
     MapMapbox,
     MapDrawControl,
     MapNavigationControl,
-    MapLayer
+    MapLayer,
+    MapSearch
+  },
+  data() {
+    return {
+      borderLayer: wms({
+        id: 'border',
+        tiles: [
+          geoserverUrl({
+            url:
+              'https://geodata.nationaalgeoregister.nl/bestuurlijkegrenzen/wms',
+            layers: 'landsgrens',
+            ...wmsLayerDefaultConfig
+          })
+        ],
+        tileSize: tileSize
+      }),
+      waterWaysLayer: wms({
+        id: 'water-ways',
+        tiles: [
+          geoserverUrl({
+            url:
+              'https://geoservices.rijkswaterstaat.nl/apps/geoserver/nwb_vaarwegen/wms',
+            layers: 'nwb_vaarwegen:vaarwegvakken',
+            ...wmsLayerDefaultConfig
+          })
+        ],
+        tileSize: tileSize
+      })
+    };
```

Comment:
You'll probably get tired of my "unnecessary-reactive-props-police" record real quick, but I have this itch against dumping stuff into `vue data` or `vuex store` unless strictly necessary. To counter reactivity, it could be `Object.freeze`, but you can also put the data directly on the component root export object. It will then be available on the template under `$options.yourUnreactiveProperty`.

Reply:
You're absolutely right. I prefer the Object.freeze method, because then you keep the seperation between data & other stuff in the component